### DISCUSSION TITLE
Improve Workflows (renovatebot)

### DISCUSSION THREAD

**Discussion by rainerleber** [sap-linuxlab/community.sap_install#421] [Ideas]
Actions should not use main. To prevent maintaining efforts dependapot or renovatebot could be used.

- Main is generally the upstream for actions which can lead into failures which possibly come with changes.
 for example main could also reflect a major change.
- Only Released Versions are stable
- It's a good idea to mention actions with `@v4` for example to get minor and patch updates but not break the behavior when a major change is introduced.

**Comment by rainerleber** [comment]
@sean-freeman @berndfinger @rhmk  @ja9fuchs

**Comment by Wabri** [comment]
Done in #610

**Comment by berndfinger** [comment]
@rainerleber Could you please look into https://github.com/sap-linuxlab/community.sap_install/pull/610 and maybe also provide more details on your suggestion regarding dependabot or renovatebot?

**Reply by kksat** [reply]
I am also interested, I did not know about renovate. I see in the documentation it also supports ansible galaxy collections. If you ca n share some insights how to use renovate for ansible this will be great and very appreciated.

**Reply by Wabri** [reply]
site: https://docs.renovatebot.com/
github action: https://github.com/marketplace/actions/renovate-bot-github-action

**Comment by Wabri** [comment]
I've done an [experiment in my fork](https://github.com/Wabri/community.sap_install/tree/experiment/discussion/421) that you can find the draft pr here #657.

This github action simply install all the dependencies and check if there are some of them that need to be update, if there are it open a issue, but if there is already an issue the communicate the same outdate dependency it will do nothing.

You can find some example here:

- this is the workflow I run -> https://github.com/Wabri/community.sap_install/actions/runs/7891026402/job/21534475374#step:3:227 if you go at the end of the `check workflow dependency` you find the output `Issue created successfully.` and this is because the action create 4 issues:
  - https://github.com/Wabri/community.sap_install/issues/14
  - https://github.com/Wabri/community.sap_install/issues/15
  - https://github.com/Wabri/community.sap_install/issues/16
  - https://github.com/Wabri/community.sap_install/issues/17
- If I run again the workflow the issues will not be created even if the issue is closed -> https://github.com/Wabri/community.sap_install/actions/runs/7891090096/job/21534686839#step:3:225

There are something that need to be setup in order to have every environment variable.

With this we can avoid the use of external workflows like renovate or dependabot, let me know what are you think about this solution!

**Reply by Wabri** [reply]
I can also create an automatic pr opener that apply the update within the action that refer to the issue already opened

**Comment by berndfinger** [comment]
I checked the outputs and the behavior. Looks good to me! I think will be useful for us.

**Reply by Wabri** [reply]
In order to move this to the main repository we need to:
- select a user that will create the issue using this workflow
- create a personal fine-grained token of the user with `Read and Write access to issues` permission
- add the value of the token to `Repository secrets` find in the `Actions secrets and variables` https://github.com/sap-linuxlab/community.sap_install/settings/secrets/actions (I dont have access to that section)
- choose the title structure, for now is like this `Dependency outdated in <requirement_file>: <package_name>:<version> -> <latest_version>`
- how the workflow is triggered, in my fork I activated manually
